### PR TITLE
[#4062] feat(iceberg-rest-server): support RegisterTable for Iceberg REST server

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
1. upgrade Iceberg version to 1.5.2
2. support register table interface for Iceberg REST server

### Why are the changes needed?

Fix: #4062 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
1. add UT and IT
2. test in local env

### CODE REVIEW

File: iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ void testLoadCatalog() {
         });
 
     Map<String, String> properties = new HashMap<>();
-    properties.put(CatalogProperties.URI, "jdbc://0.0.0.0:3306");
```
Comment (FANNG1) [inline iceberg/iceberg-common/src/test/java/org/apache/gravitino/iceberg/common/utils/TestIcebergCatalogUtil.java:63]: the new Iceberg version will check the correctness of jdbc uri

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergTableOperations.java [hunk 2]
```diff
@@ -107,10 +108,10 @@ private Response doLoadTable(String name) {
   }
 
   private Response doUpdateTable(String name, TableMetadata base) {
-    MetadataUpdate addSchema = new AddSchema(newTableSchema, base.lastColumnId());
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/rest/TestIcebergTableOperations.java:110]: API changes for new Iceberg Version

**Comment by FANNG1** [comment]
@jerryshao please help to review when you have time

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java [hunk 3]
```diff
@@ -526,4 +519,42 @@ void testSnapshot() {
...
+            "CREATE TABLE %s.%s (id bigint COMMENT 'unique id',data string) using iceberg",
+            registerDB, registerTableName));
+    sql(String.format("INSERT INTO %s.%s VALUES (1, 'a')", registerDB, registerTableName));
+
+    // get metadata location
+    List<String> metadataLocations =
+        convertToStringList(
+            sql(
+                String.format(
+                    "select file from %s.%s.metadata_log_entries", registerDB, registerTableName)),
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java:540]: Can you please make all the SQL keywords uppercase.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java [hunk 4]
```diff
@@ -526,4 +519,42 @@ void testSnapshot() {
...
+            "CALL rest.system.register_table(table => 'iceberg_rest_table_test.register_foo2', metadata_file=> '%s')",
+            metadataLocation);
+    sql(register);
+
+    Map<String, String> result =
+        convertToStringMap(sql("select * from iceberg_rest_table_test.register_foo2"));
+    Assertions.assertEquals(ImmutableMap.of("1", "a"), result);
+
+    // insert other data
+    sql(" INSERT INTO iceberg_rest_table_test.register_foo2 VALUES (2, 'b')");
```
Comment (jerryshao) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java:556]: Remove the whitespace before "INSERT".

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java (hunk 3 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java]: done

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java (hunk 4 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRESTServiceIT.java]: done

**Comment by FANNG1** [comment]
@jerryshao @jerqi , comments are addressed, could you help to review again?

**Review by jerryshao — APPROVED** [review]

**Pull request merged**